Logic/src/Messages/CreateConversation.cs

```diff
@@ -0,0 +1,102 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+using Microsoft.AspNetCore.Http;
+
+namespace GoRideShare
+{
+    public class CreateConversation
+    {
+        private readonly ILogger<CreateConversation> _logger;
+        private readonly string? _baseApiUrl;
+        private readonly IHttpRequestHandler _httpRequestHandler;
+
+        public CreateConversation(ILogger<CreateConversation> logger, IHttpRequestHandler httpRequestHandler)
+        {
+            _logger = logger;
+            _httpRequestHandler = httpRequestHandler;
+            _baseApiUrl = Environment.GetEnvironmentVariable("BASE_API_URL");
+        }
+
+        // This function is triggered by an HTTP POST request to create a new post
+        [Function("CreateConversation")]
+        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous, "post")] HttpRequest req)
+        {
+            // Read the user ID and the db token from the headers
+            if (!req.Headers.TryGetValue("X-User-ID", out var userId))
+            {
+                return new BadRequestObjectResult("Missing the following header: \'X-User-ID\'.");
+            }
+            if (!req.Headers.TryGetValue("X-Db-Token", out var db_token))
+            {
+                return new BadRequestObjectResult("Missing the following header \'X-Db-Token\'.");
+            }
```

Comment:
This block is repeated everywhere, I think we can make it into function. Good for testing and reusability:

```
// Method that validates headers, outputs the userID and dbToken, returns exception if headers  missing, null if headers are good
private IActionResult ValidateHeaders(IHeaderDictionary headers, out string userId, out string dbToken)
{
    userId = string.Empty;
    dbToken = string.Empty;

    // Check for X-User-ID  and X-DbToken headers
    if (!headers.TryGetValue("X-User-ID", out var userIdValue))
    {
        return new BadRequestObjectResult("Missing the following header: 'X-User-ID'.");
    }
    userId = userIdValue.ToString();

    if (!headers.TryGetValue("X-Db-Token", out var dbTokenValue))
    {
        return new BadRequestObjectResult("Missing the following header: 'X-Db-Token'.");
    }
    dbToken = dbTokenValue.ToString();

    return null; // All headers are valid
}
```

Put this in `Run` instead of the repeated if statements
```
    var validationResult = ValidateHeaders(req.Headers, out string userId, out string dbToken);
    
    // If validation result is not null, return the bad request result
    if (validationResult != null)
    {
        return validationResult;
    }

    // if here, the headers are good
```

Reply:
I don't think this will work since `out var dbTokenValue` is then lost by the function.
If we create a new function to handle this, it needs to return both headers as variables.
With that restriction I don't think we can fix this without using weird tuple/object return types.

Reply:
You didn't look at the suggested code carefully enough. The function `ValidateHeaders` signature  includes `out string userID, out string dbToken`, This means you can call it using `ValidateHeaders(req.Headers, out string userId, out string dbToken);` and the `userID and dbToken` are not lost. 

It's a C# feature. A function can return one thing, but output another. Here we are returning the exception and outputting the variables.

Reply:
Oh that's neat, I didn't realize it works that way. I'll switch it around now.